evaluation.md

```diff
@@ -5,15 +5,39 @@
 _Describe each change from your ideal syntax to the syntax you implemented, and
 describe_ why _you made the change._
 
+Many of the changes had to do with limitations with Scala (or at least what I could accomplish in it in a reasonable about of time). In fact, most of the changes can be described as making my "ideal" language more Scala-y. For example, comments are now made with a double slash instead of a pound sign. I also had to move away from a more Pythonic syntax to Scala's own syntax. So, state blocks don't start with a colon and are instead surrounded by curly brackets. Scala line indentation also applies. Since we have to use Scala for the internal DSL, we also have to run the code within an object that extends JFXApp and incldue import statements, which takes away from the clean interface of the original language. Another notable change is the addition of parentheses everywhere and the lack of spaces between many words. These were also forced by Scala's own contrains and the time I wanted to spend on this assignment.
+
+One feature that I _added_ to the internal DSL was the "WhenIn" keyword to indicate that you were entering a state block. I spent a long time trying to get it working without a keyword beforehand, but implicit conversion and by-name parameters apparently do not mix well in Scala. Not even my hacky workarounds succeeded (I tried returning a function to do the same job as a by-name parameter like in my final WhenIn implementation). While I was somewhat forced to create WhenIn, it was actually kind of a nice addition and some people might actually prefer using it.
+
+Other changes include requiring quotes around state names, which can actually be nice if your text editor has syntax highlighting, and eliminating some spaced between words for compactness and ease of implementation. The compact names also helped make the automatic completion in my text editor more useful. One feature I ended up droping was the ability to group directions when specifying constraints. In the original design, it was legal to say "If cannot move left or right or down...". This would not only have been more work for me, but could also annoy programmers if they needed even more parethesis to make this work (an alternative solution would be to create an object for every combinatin of directions, which could significantly affect compile times as there would be about 128 unique objects).
+
+Finally, one last feature I added was a "runOnMaze" statement that allows the user to specify the map they want to use. As before, this was also kind of forced as I needed some way to know that all the rules had been specified and which map to load. This is something I didn't think about much in the design phase and would have been good to incorporate as it's currently ambiguous what map the code should be run on.
+
```

Comment:
Huh, we didn't think about this, we just put it in when we ran our program. I like that you put some thought into it!